aten/src/ATen/native/native_functions.yaml

```diff
@@ -487,6 +487,10 @@
 - func: all.out(Tensor self, int dim, bool keepdim=False, *, Tensor(a!) out) -> Tensor(a!)
   device_check: NoCheck   # TensorIterator
   structured: True
+  precomputed:
+    elements: int64_t dim_
```

Comment:
I can't remember: did we decide we were going to have elements be JIT schema types or not?

Reply:
No, I think we said we need them to be C++ types because of something CUDA related that i can't remember. I did some Googling and found this post on StackOverflow that states that [operations on 64-bit integers are synthesized using multiple 32-bit operations](https://stackoverflow.com/questions/16845644/64-bit-number-support-in-cuda/16845794), so there could be performance benefits to using 32-bit integers. In the case of `avg_pool2d` for example, the precomputed elements are used to calculate `int` loop bounds, so that explanation kind of makes sense.

However, that post is 8 years old, so it could be out of date.

Reply:
Ah yes, I remember this. But I think I also said something like, it might still be OK to store these as 64-bit and then force the CUDA kernel to truncate them (and it wouldn't be necessary on CPU). Not clear if other kernels would have similar problems or not tho...

Reply:
Okay, so the CPU kernel uses `int64_t` for the actual loop variables derived from these `int` parameters:

https://github.com/pytorch/pytorch/blob/855e8f2b179e3cb7c0e11f9c7757acd610a2a6b4/aten/src/ATen/native/cpu/AvgPoolKernel.cpp#L51-L54

but the CUDA kernel uses `int`:

https://github.com/pytorch/pytorch/blob/855e8f2b179e3cb7c0e11f9c7757acd610a2a6b4/aten/src/ATen/native/cuda/AveragePool2d.cu#L37-L40

so I think your proposal makes sense. Let me try replacing the C++ types in `native_functions.yaml` with JIT types and seeing how it goes.